src/utils/fetchWithDelay.test.ts

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { describe, it, expect } from "@jest/globals";
+import { faker } from "@faker-js/faker/.";
+import fetchWithDelay from "./fetchWithDelay";
+
+describe("fetchWithDelay", () => {
+  it("resolves after the delay", async () => {
+    global.fetch = jest.fn().mockResolvedValueOnce({
+      json: () => Promise.resolve({ success: true }),
+    });
+    const response = await fetchWithDelay(`${faker.internet.url()}/api/test`, {
+      delay: 750,
+    });
+    const data = await response.json();
+    expect(data).toStrictEqual({ success: true });
+  });
+
+  it("throws an error", async () => {
+    global.fetch = jest.fn().mockRejectedValueOnce("fetch failed");
+    try {
+      expect(
+        await fetchWithDelay(`${faker.internet.url()}/api/test`, {
+          delay: 750,
+        }),
+      ).toThrow("fetch failed");
+    } catch (_) {
```

Comment:
Hm, I'm surprised this is necessary... does the `expect(...).toThrow()` throw in this case? Or is this for some other unexpected error? Might want to log it just in case something unrelated happens.

Reply:
Now you are mentioning it — I was surprised as well. I’ve updated the test to work as expected: [dce81dd](https://github.com/mozilla/blurts-server/pull/5395/commits/dce81ddc7c9da0b45d42388ebadc53e97f502f42).